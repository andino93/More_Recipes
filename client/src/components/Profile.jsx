import React from 'react';
import ReactDOM from 'react-dom';
import HeaderComponent from './partials/Header';
import SignInComponent from './partials/User/SignIn';
import SignUpComponent from './partials/User/SignUp';
import FooterComponent from './partials/Footer';
import UserPanelComponent from './partials/UserPanel';

/**
 *
 *
 * @export
 * @class ProfileComponent
 * @extends {React.Component}
 */
export default class ProfileComponent extends React.Component {
    render() {
        return (
            <div>
                <HeaderComponent/>
                <SignInComponent/>
                <SignUpComponent/>
                <div className="wrapper">
                    <ol
                        className="breadcrumb container mt-50 mb-10 col-md-9 mx-auto bg-white shadow-lite">
                        <li className="breadcrumb-item">
                            <a href="#">Home</a>
                        </li>
                        <li className="breadcrumb-item">
                            <a href="#">User</a>
                        </li>
                        <li className="breadcrumb-item active">Profile</li>
                    </ol>
                    <div className="mb-20 col-md-9 mx-auto recipe-details-container">
                        <div className="row">
                            <UserPanelComponent/>
                            <div
                                className="simplebox blade col-12 col-sm-8 col-md-8 col-lg-8 p-10"
                                style={{
                                boxShadow: 'none'
                            }}>
                                <form className="mt-15">
                                    <div className="form-group">
                                        <input
                                            type="email"
                                            className="form-control"
                                            id="inputEmail1"
                                            aria-describedby="emailHelp"
                                            placeholder="Email"
                                            value="mohammedisioye@gmail.com"
                                            readOnly/>
                                    </div>
                                    <div className="form-group">
                                        <input
                                            type="text"
                                            className="form-control"
                                            id="inputUserName"
                                            aria-describedby="userNameHelp"
                                            placeholder="Username"
                                            value="@mohzak"
                                            readOnly/>
                                    </div>
                                    <div className="form-group">
                                        <input
                                            type="text"
                                            className="form-control"
                                            id="inputFullName"
                                            aria-describedby="fullNameHelp"
                                            placeholder="Full name"/>
                                    </div>

                                    <div className="form-group">
                                        <input
                                            type="text"
                                            className="form-control"
                                            id="inputMobile"
                                            aria-describedby="mobileHelp"
                                            placeholder="Mobile Number"/>
                                    </div>
                                    <button type="submit" className="btn btn-outline-warning">Update</button>
                                </form>
                                <div className="row">
                                    <div className="col-4 col-md-4 col-sm-4">
                                        <hr/>
                                    </div>
                                    <div className="col-4 col-md-4 col-sm-4 text-center">
                                        <h5 className="text-muted">Change Password</h5>
                                    </div>
                                    <div className="col-4 col-md-4 col-sm-4">
                                        <hr/>
                                    </div>
                                </div>
                                <form className="mt-15">
                                    <div className="form-group">
                                        <input
                                            type="password"
                                            className="form-control"
                                            id="inputPassword"
                                            placeholder="Current Password"/>
                                    </div>
                                    <div className="form-group">
                                        <input
                                            type="password"
                                            className="form-control"
                                            id="inputPassword2"
                                            placeholder="New Password"/>
                                    </div>
                                    <div className="form-group">
                                        <input
                                            type="password"
                                            className="form-control"
                                            id="inputPassword3"
                                            placeholder="Confirm New Password"/>
                                    </div>
                                    <button type="submit" className="btn btn-success">Submit</button>
                                </form>
                            </div>
                        </div>
                    </div>
                </div>
                <FooterComponent/>
            </div>
        )
    }
}