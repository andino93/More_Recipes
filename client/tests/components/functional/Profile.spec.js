import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import mockData from "../../__mocks__/mockData";
import localStorageMock from '../../__mocks__/localStorageMock';
import Profile from '../../../src/components/Profile.jsx';

configure({ adapter: new Adapter() });

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

describe('<Profile>', () => {
  const store = mockStore({});
  const connectedProfile = mountWrap(Profile, store, {
    match: {}
  });
  it('should render the Profile Component', (done) => {
    expect(toJson(connectedProfile)).toMatchSnapshot();
    done();
  });
  it('should check for componentWillReceiveProps', () => {
    connectedProfile.find('Profile').instance().componentWillReceiveProps({
      getUserState: {
        responseType: 'GET_USER_PROFILE_SUCCESS',
      },
    });
  });
});