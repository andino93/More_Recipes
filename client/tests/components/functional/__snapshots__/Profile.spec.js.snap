// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> should render the Profile Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Profile)
        match={Object {}}
      >
        <Profile
          getUser={[Function]}
          match={Object {}}
        >
          <div>
            <Connect(Header)>
              <Header
                logoutAction={[Function]}
              >
                <div>
                  <nav
                    className="navbar navbar-expand-sm navbar-light bg-navbar fixed-top"
                  >
                    <Link
                      className="navbar-brand"
                      replace={false}
                      to="/"
                    >
                      <a
                        className="navbar-brand"
                        href="/"
                        onClick={[Function]}
                      >
                        More Recipes
                      </a>
                    </Link>
                    <button
                      aria-controls="navbarSupportedContent"
                      aria-expanded="false"
                      aria-label="Toggle navigation"
                      className="navbar-toggler"
                      data-target="#navbarSupportedContent"
                      data-toggle="collapse"
                      type="button"
                    >
                      <span
                        className="navbar-toggler-icon"
                      />
                    </button>
                    <div
                      className="collapse navbar-collapse"
                      id="navbarSupportedContent"
                    >
                      <ul
                        className="navbar-nav ml-auto"
                      >
                        <li
                          className="nav-item dropdown"
                        >
                          <Link
                            aria-expanded="false"
                            aria-haspopup="true"
                            className="nav-link dropdown-toggle"
                            data-toggle="dropdown"
                            id="navbarDropdown"
                            replace={false}
                            role="button"
                            to="#"
                          >
                            <a
                              aria-expanded="false"
                              aria-haspopup="true"
                              className="nav-link dropdown-toggle"
                              data-toggle="dropdown"
                              href="#"
                              id="navbarDropdown"
                              onClick={[Function]}
                              role="button"
                            >
                              My Account
                            </a>
                          </Link>
                          <div
                            aria-labelledby="navbarDropdown"
                            className="dropdown-menu dropdown-menu-right mr-20"
                          >
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/profile"
                            >
                              <a
                                className="dropdown-item"
                                href="/profile"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-user-circle"
                                />
                                 My Profile
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/user-recipes"
                            >
                              <a
                                className="dropdown-item"
                                href="/user-recipes"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-list"
                                />
                                 My Recipes
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/my-favorite"
                            >
                              <a
                                className="dropdown-item"
                                href="/my-favorite"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-star"
                                />
                                 My Favorites
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/addRecipe"
                            >
                              <a
                                className="dropdown-item"
                                href="/addRecipe"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-plus"
                                />
                                 Add Recipes
                              </a>
                            </Link>
                            <div
                              className="dropdown-divider"
                            />
                            <Link
                              className="dropdown-item"
                              onClick={[Function]}
                              replace={false}
                              to="#"
                            >
                              <a
                                className="dropdown-item"
                                href="#"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-sign-out"
                                />
                                Log out
                              </a>
                            </Link>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </nav>
                </div>
              </Header>
            </Connect(Header)>
            <div
              className="wrapper"
            >
              <ol
                className="breadcrumb container mt-50 col-10 col-md-10 col-lg-10 mx-auto bg-white shadow-lite"
              >
                <li
                  className="breadcrumb-item"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="breadcrumb-item"
                >
                  <a
                    href="#"
                  >
                    User
                  </a>
                </li>
                <li
                  className="breadcrumb-item active"
                >
                  Profile
                </li>
              </ol>
              <div
                className="center-content"
              >
                <img
                  alt="avatar"
                  className="avatar text-center circulate circulate-xlg"
                  src="/assets/images/no-avatar.png"
                />
              </div>
              <div
                className="mb-20 col-md-10 mx-auto recipe-details-container"
              >
                <div
                  className="row"
                >
                  <div
                    className="simplebox blade col-12 col-sm-12 col-md-12 col-lg-12 p-10"
                    style={
                      Object {
                        "boxShadow": "none",
                      }
                    }
                  >
                    <ul
                      className="list-group list-group-flush"
                    >
                      <li
                        className="list-group-item"
                      >
                        <p>
                          <b>
                            First Name: 
                          </b>
                        </p>
                      </li>
                      <li
                        className="list-group-item"
                      >
                        <p>
                          <b>
                            Last Name: 
                          </b>
                        </p>
                      </li>
                      <li
                        className="list-group-item"
                      >
                        <p>
                          <b>
                            Username: 
                          </b>
                        </p>
                      </li>
                      <li
                        className="list-group-item"
                      >
                        <p>
                          <b>
                            Email:
                          </b>
                           
                        </p>
                      </li>
                    </ul>
                     
                    <br />
                    <h5
                      className="text-info text-center"
                    >
                      You joined this platform 
                       
                      Invalid date
                    </h5>
                  </div>
                </div>
              </div>
            </div>
            <FooterComponent>
              <footer
                className="p-10 well well-sm m-0 shadow-lite"
              >
                <div
                  className="text-center text-muted"
                >
                  <p
                    className="bold"
                  >
                    © 2017 More Recipes
                  </p>
                </div>
              </footer>
            </FooterComponent>
          </div>
        </Profile>
      </Connect(Profile)>
    </Router>
  </BrowserRouter>
</Provider>
`;
